ruskeyverbs/views.py

```diff
@@ -26,19 +51,618 @@ def index(request):
+    """
+    view for multiple choice quiz
+    """
+    example_inst = get_object_or_404(Example, pk=pk)
+    russian_text_list = strip_punct_lower(example_inst.russian_text,
+                                          stressed=True).split()
+    unstressed_russian_text_list = strip_punct_lower(
+        example_inst.russian_text).split()
+    """
+    test each word of the sentence to see which one is a form of the verb
+    being studied and determine which form it is
+    """
+    print(request.session)
+    field_names = Verb._meta.get_fields()
+    for i in range(len(russian_text_list)):
+        for field in field_names:
+            try:
+                verb_form = getattr(example_inst.verb, field.name)
+                if verb_form == russian_text_list[i]:
+                    target_field = field
+                    answer = unstressed_russian_text_list[i]
+                    stressed_answer = russian_text_list[i]
+            except AttributeError:
+                pass
+                # pass to get around many to one field returned by get_fields()
+    # create a list of the pks of 3 other random verbs and randomize it
+    choice_pks = [int(example_inst.verb.pk)]
+    while len(choice_pks) < 4:
+        random_pk = randint(1, Verb.objects.count())
+        while random_pk in choice_pks:
+            random_pk = randint(1, Verb.objects.count())
+        choice_pks.append(random_pk)
+    shuffle(choice_pks)
+    request.session['mcchoices'] = choice_pks
+    request.session['target_field'] = target_field.name
+    choice_list = []  # list of tuples that will be passed to the form
+    # grab the right verb form for the other verbs
+    for i in choice_pks:
+        choice_list.append(
+            (getattr(Verb.objects.get(pk=i), target_field.name).replace(
+                chr(769), ''),
+             getattr(Verb.objects.get(pk=i), target_field.name).replace(
+                 chr(769), ''))
+            )
+    quiz_text = example_inst.russian_text.replace(stressed_answer,
+                                                  '___________')
+    quiz_text = quiz_text.replace(stressed_answer.capitalize(),
+                                  '____________')
+    try:
+        print(request.session['progress'])
+    except KeyError:
+        request.session['progress'] = 0
+    try:
+        request.session['quiz_state'] += 1
+        request.session.modified = True
+    except KeyError:
+        request.session['quiz_state'] = 0
+    form = MultipleChoiceForm(choice_list)
+    if request.session['quiz_state'] == 1:
+        return render(
+            request,
+            'ruskeyverbs/multiple_choice.html',
+            context={
+                'quiz_text': quiz_text,
+                'form': form,
+                'pk': pk,
+                'file': example_inst.example_audio,
+                'english_text': example_inst.translation_text
+            }
+            )
+    else:
+        raise Http404
```

Comment:
Do you really have to do all the work above this in this method just to determine that you need to return a 404 here?